Content.Client/_DV/CosmicCult/UI/Monument/MonumentMenu.xaml.cs

```diff
@@ -0,0 +1,190 @@
+    }
+
+    // Update all the glyph buttons
+    private void UpdateGlyphs()
+    {
+        GlyphContainer.RemoveAllChildren();
+        foreach (var glyph in _glyphPrototypes)
+        {
+            var boxContainer = new BoxContainer();
+            var unlocked = _unlockedGlyphProtoIds.Contains(glyph.ID);
+            var button = new Button
+            {
+                HorizontalExpand = true,
+                StyleClasses = { StyleBase.ButtonSquare },
+                ToolTip = Loc.GetString(glyph.Tooltip),
+                Group = _glyphButtonGroup,
+                Pressed = glyph.ID == _selectedGlyphProtoId,
+                Disabled = !unlocked,
+                Modulate = !unlocked ? Color.Gray : Color.White,
+            };
+            button.OnPressed += _ => _selectedGlyphProtoId = glyph.ID;
+            var glyphIcon = new TextureRect
+            {
+                Texture = _sprite.Frame0(glyph.Icon),
+                TextureScale = new Vector2(2f, 2f),
+                Stretch = TextureRect.StretchMode.KeepCentered,
+            };
+            boxContainer.AddChild(button);
+            button.AddChild(glyphIcon);
+            GlyphContainer.AddChild(boxContainer);
+        }
+    }
+
+    // Update all the influence thingies
+    private void UpdateInfluences(MonumentBuiState state)
+    {
+        InfluencesContainer.RemoveAllChildren();
+
+        var influenceUIBoxes = new List<InfluenceUIBox>();
+        foreach (var influence in _influencePrototypes)
+        {
+            var uiBoxState = GetUIBoxStateForInfluence(influence, state);
+            var influenceBox = new InfluenceUIBox(influence, uiBoxState);
+            influenceUIBoxes.Add(influenceBox);
+            influenceBox.OnGainButtonPressed += () => OnGainButtonPressed?.Invoke(influence.ID);
+        }
+
+        //sort the list of UI boxes by state (locked -> owned -> not enough entropy -> enough entropy)
+        //then sort alphabetically within those categories
+        influenceUIBoxes = influenceUIBoxes.OrderBy(box => box.State)
+            .ThenBy(box => box.Proto.ID)
+            .ToList();
+
+        foreach (var box in influenceUIBoxes)
+        {
+            InfluencesContainer.AddChild(box);
+        }
+    }
+
+    private InfluenceUIBox.InfluenceUIBoxState GetUIBoxStateForInfluence(InfluencePrototype influence, MonumentBuiState state)
+    {
+        if (!_ent.TryGetComponent<CosmicCultComponent>(_player.LocalEntity, out var cultComp))
+            return InfluenceUIBox.InfluenceUIBoxState.Locked; //early return with locked if there's somehow no cult comp
+
+        var unlocked = cultComp.UnlockedInfluences.Contains(influence.ID);
+        var owned = cultComp.OwnedInfluences.Any(ownedProtoId => ownedProtoId.Id.Equals(influence.ID));
+
+        //more verbose than it needs to be, but it reads nicer
+        if (owned)
+            return InfluenceUIBox.InfluenceUIBoxState.Owned;
+
+        if (unlocked)
```

Comment:
Avoid calling functions when you're not gonna check the value, also don't use `Any` to check for `Contains`
Additionally, there's an `implicit operator ProtoId<T>(T proto)` implementation, so no need to `.ID`
```suggestion
        var owned = cultComp.OwnedInfluences.Contains(influence);
        if (owned)
            return InfluenceUIBox.InfluenceUIBoxState.Owned;

        var unlocked = cultComp.UnlockedInfluences.Contains(influence);
        if (unlocked)
```